File: MarsalaApi/src/Common/Model/Order.cs
using System.Collections.Generic;

namespace Common.Model
{
	public class Order
	{
		public string UserName { get; set; }
		public string Salad { get; set; }
		public string Soup { get; set; }
		public string MainCourse { get; set; }
		public string Drink { get; set; }
		public PaymentMethod PaymentMethod { get; set; }
		public ICollection<string> Snacks { get; set; }
	}

	public enum PaymentMethod
	{
		Cash = 1,
		Card = 2
	}
}

File: MarsalaApi/src/MarsalaWeb/angularApp/app/Models/order.ts
export interface IOrder {
    UserName: string;
    PaymentMethod: PaymentMethod;
    Salad: string;
    Soup: string;
    MainCourse: string;
    Drink: string;
    Snacks: string[];
}

export class Order implements IOrder {
    UserName: string;
    PaymentMethod: PaymentMethod;
    Salad: string;
    Soup: string;
    MainCourse: string;
    Drink: string;
    Snacks: string[] = [];
}

export enum PaymentMethod {
    Cash = 1,
    Card = 2
}

export interface ICourse {
    name: string;
    count: number;
}

export interface ISummary {
    orders: IOrder[];
    orderText: string;
    Salad: ICourse[];
    Soup: ICourse[];
    MainCourse: ICourse[];
    Drink: ICourse[];
    Snacks: ICourse[];
}

export class Summary {
    orders: IOrder[] = [];
    orderText: string = "";
    Salad: ICourse[] = [];
    Soup: ICourse[] = [];
    MainCourse: ICourse[] = [];
    Drink: ICourse[] = [];
    Snacks: ICourse[] = [];
}

export interface IOrderSentStatus {
    SenderName: string;
    StatusText: string;
    IsSuccess: boolean;
    SentDate: Date;
}